Preact example doesn't start or build

# Examples bug report

## Example name

`using-preact`

## Describe the bug

When trying to run `yarn dev` the example breaks with the error:

```
$ node server.js
/home/rafael/projects/next.js/examples/using-preact/node_modules/next/dist/next-server/lib/amp-context.js:11
exports.AmpStateContext = React.createContext({});
                                ^

TypeError: React.createContext is not a function
    at Object.<anonymous> (/home/rafael/projects/next.js/examples/using-preact/node_modules/next/dist/next-server/lib/amp-context.js:11:33)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/rafael/projects/next.js/examples/using-preact/node_modules/next/dist/next-server/lib/head.js:8:23)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
error Command failed with exit code 1.
```

## To Reproduce

1. Go to `examples/using-preact`
2. Run `yarn install && yarn dev`
4. See error

## Expected behavior

The app should start successfully

## System information

- OS: Ubuntu 19.04
- Version of Next.js: 9.1.1

## Additional context

`createContext` is already built-in on the latest version of Preact (Preact X), it could be as simple as updating it, but then comes the second problem: you cannot run `yarn build` without `react` and `react-dom` installed in the project:

```
$ next build
The module 'react' was not found. Next.js requires that you include it in 'dependencies' of your 'package.json'. To add it, run 'npm install --save react'
The module 'react-dom' was not found. Next.js requires that you include it in 'dependencies' of your 'package.json'. To add it, run 'npm install --save react-dom'
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'react'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/rafael/projects/next.js/examples/using-preact/node_modules/next/dist/bin/next:29:13)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
error Command failed with exit code 1.
```

Does the project need to have `react` and `react-dom` installed to be built, even if it is already using Preact, or is there a way to alias these modules to the build step like we do in `next.config.js`?

--- a/examples/using-preact/README.md
+++ b/examples/using-preact/README.md
@@ -1,4 +1,4 @@
-# Hello World example
+# Preact example
 
 ## How to use
 
@@ -43,4 +43,5 @@ This example uses [Preact](https://github.com/developit/preact) instead of React
 
 Here's how we did it:
 
-- Use `next.config.js` to customize our webpack config to support [preact-compat](https://github.com/developit/preact-compat)
+- Use `next.config.js` to customize our webpack config by aliasing React to `preact/compat`
+- Use `server.js` to make our server use Preact by aliasing React to `preact/compat`

--- a/examples/using-preact/next.config.js
+++ b/examples/using-preact/next.config.js
@@ -1,3 +1,13 @@
-const withPreact = require('@zeit/next-preact')
+module.exports = {
+  webpack: function (config) {
+    config.resolve.alias = {
+      ...config.resolve.alias,
+      react: 'preact/compat',
+      react$: 'preact/compat',
+      'react-dom': 'preact/compat',
+      'react-dom$': 'preact/compat'
+    }
 
-module.exports = withPreact()
+    return config
+  }
+}

--- a/examples/using-preact/package.json
+++ b/examples/using-preact/package.json
@@ -7,10 +7,14 @@
     "start": "NODE_ENV=production node server.js"
   },
   "dependencies": {
-    "@zeit/next-preact": "0.1.0",
+    "module-alias": "2.2.2",
     "next": "latest",
-    "preact": "8.2.7",
-    "preact-compat": "3.18.0"
+    "preact": "10.0.0",
+    "preact-render-to-string": "5.0.7"
   },
-  "license": "ISC"
+  "license": "ISC",
+  "devDependencies": {
+    "react": "16.10.2",
+    "react-dom": "16.10.2"
+  }
 }

--- a/examples/using-preact/server.js
+++ b/examples/using-preact/server.js
@@ -1,7 +1,11 @@
 const port = parseInt(process.env.PORT, 10) || 3000
 const dev = process.env.NODE_ENV !== 'production'
+const moduleAlias = require('module-alias')
 
-require('@zeit/next-preact/alias')()
+moduleAlias.addAliases({
+  react: 'preact/compat',
+  'react-dom': 'preact/compat'
+})
 
 const { createServer } = require('http')
 const { parse } = require('url')

